peekaboo/toolbox/cuckoo.py

```diff
@@ -314,11 +323,10 @@ def do(self):
                     # ignore and retry on next polling run
                     continue
 
-                # but fail hard if we get invalid stuff
                 if "task" not in job or "status" not in job["task"]:
                     logger.error("Invalid JSON structure from Cuckoo REST "
                                  "API: %s", job)
-                    return 1
+                    continue
```

Comment:
Yeah, that seems a bit drastic at first but there's reason to my madness:
1. Realistically, getting some invalid JSON from the REST API is no worse that it being unreachable - we can and IMO should try to continue instead of dying.
2. We aren't monitoring any of our threads. So they must not die. It's the same with the workers and all the others. Alternatively we'd need a monitoring loop in the main thread, waiting for any thread to exit (couldn't find a threading function to do that) and then shut down all the others. This would actually be useful for the workers but due to point 1 above IMO not for job trackers.